docs/source/docker/getting_started/hardware_recommendations.rst

```diff
@@ -0,0 +1,82 @@
+.. _ref-hardware-recommendations:
+
+Hardware recommendations
+========================
+
+Usually, the Lightly worker is run on a cloud machine,
+which is spin up specifically to run the docker.
+Our recommendations on the hardware configuration of this machine are
+based on three criteria:
+
+- speed: The worker should process your dataset as quickly a possible.
+- cost-effectiveness: The machine should be cheap to rent.
+- stability: The worker should not crash because it runs out of memory.
+
+As a starting point, we recommend a machine with the following configuration:
+
+- 8 vCPUs
+- 30-32 GB of system memory
+- a NVIDIA-GPU with 16GB of video memory, e.g. the T4 GPU.
+
+If you read the data from a local hard disk, make sure that the connection is fast,
+ideally faster than 100MB/s.
+
+If you read the data from a cloud bucket instead, make sure that
+the cloud bucket is in the same region as the compute machine
+and the download speed is fast, ideally faster than 100MB/s.
+
+Keep the configuration option `lightly.loader.num_workers` at the default (-1),
+which will set it to the number of vCPUs on your machine.
+
+The best machine hardware configuration is highly dependent
+on the type of data you have and how you configure the worker.
+Thus we recommend to adapt it to your needs.
+
+Finding the compute speed bottleneck
+------------------------------------
+
+Usually, the compute speed is limited by one of three potential bottlenecks:
+
+- data read speed: I/O
+- CPU
+- GPU
+
+Different steps of the Lightly worker use these resources to a different extent.
+Thus the bottleneck changes throughout the run.
+The GPU is used at three steps:
+
+- training an embedding model (optional step)
+- pretagging your dataset (optional step)
+- embedding your dataset
+
+The I/O and CPUs are used at the former 3 steps and also used at the other steps that may take longer:
+
+- initializing the dataset
+- corruptness check (optional step)
+- dataset dumping & upload (optional step)
+
+Before updating one or all of resources, we recommend finding out the current bottleneck:
```

Comment:
Before updating one or all of resources -> Before changing the hardware configuration of your compute instance, we recommend to first determine the bottleneck by monitoring it: